src/rgw/rgw_rest_role.cc

```diff
@@ -107,8 +97,36 @@ int RGWRestRole::verify_permission(optional_yield y)
     return -EACCES;
   }
 
+  return 0;
+}
+
+int RGWRestRole::init_processing(optional_yield y)
+{
+  string role_name = s->info.args.get("RoleName");
+  std::unique_ptr<rgw::sal::RGWRole> role = driver->get_role(role_name,
+                                                             s->user->get_tenant());
+  if (int ret = role->get(s, y); ret < 0) {
+    if (ret == -ENOENT) {
+      return -ERR_NO_ROLE_FOUND;
+    }
+    return ret;
+  }
   _role = std::move(role);
+  return 0;
+}
 
+int RGWCreateRole::init_processing(optional_yield y)
+{
+  return 0;
+}
+
+int RGWGetRole::init_processing(optional_yield y)
+{
+  return 0;
+}
+
+int RGWListRoles::init_processing(optional_yield y)
+{
```

Comment:
looks good, but please group these functions together with their classes

and it's worth adding a comment like
```c++
  return 0; // avoid calling RGWRestRole::init_processing()
```

Reply:
Done!